src/common/FileSystem.cpp

```diff
@@ -2155,9 +2157,14 @@ bool IsInitialized()
 // Add a pak to the list of available paks
 static void AddPak(pakType_t type, Str::StringRef filename, Str::StringRef basePath)
 {
+	size_t suffixLen;
 	std::string fullPath = Path::Build(basePath, filename);
 
-	size_t suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+	if (!UseLegacyPaks()) {
+		suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+	} else {
+		suffixLen = type == pakType_t::PAK_DIR ? strlen(LEGACY_PAK_DIR_EXT) : strlen(LEGACY_PAK_ZIP_EXT);
```

Comment:
that part makes no sense at all and only works because `strlen(pk3) == strlen(dpk)`, I'm fixing this.